File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/apply_signed_char_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 12;
BEGIN { use_ok('apply_signed_char') }
require_ok('apply_signed_char');

# adapted from ../java/apply_signed_char_runme.java

my $smallnum = -127;
is(apply_signed_char::CharValFunction($smallnum), $smallnum);
is(apply_signed_char::CCharValFunction($smallnum), $smallnum);
is(apply_signed_char::CCharRefFunction($smallnum), $smallnum);

$apply_signed_char::globalchar = $smallnum;
is($apply_signed_char::globalchar, $smallnum);
is($apply_signed_char::globalconstchar, -110);

my $d = new apply_signed_char::DirectorTest();
is($d->CharValFunction($smallnum), $smallnum);
is($d->CCharValFunction($smallnum), $smallnum);
is($d->CCharRefFunction($smallnum), $smallnum);

$d->{memberchar} = $smallnum;
is($d->{memberchar}, $smallnum);
is($d->{memberconstchar}, -112);

File: support_tools/swigwin-3.0.12/Examples/test-suite/ruby/apply_signed_char_runme.rb
#!/usr/bin/env ruby
#
# Put script description here.
#
# 
# 
# 
#

require 'swig_assert'
require 'apply_signed_char'

include Apply_signed_char


['CharValFunction', 'CCharValFunction', 'CCharRefFunction'].each do |m|
  [ 3, -3 ].each do |v|
    val = send( m, v )
    swig_assert( "v == val", binding, "for #{m}")
  end
end

{ 'globalchar' => -109,
  'globalconstchar' => -110,
}.each do |k,v|
  val = Apply_signed_char.send( k )
  swig_assert( "v == val", binding, "for #{k}")
end


a = DirectorTest.new

['CharValFunction', 'CCharValFunction', 'CCharRefFunction'].each do |m|
  [ 3, -3 ].each do |v|
    val = a.send( m, v )
    swig_assert( "v == val", binding, "for DirectorTest.#{m}")
  end
end

{ 'memberchar' => -111,
  'memberconstchar' => -112,
}.each do |k,v|
  val = a.send( k )
  swig_assert( "v == val", binding, "for #{k}")
end
